.cargo/config

```diff
@@ -1,3 +1,32 @@
 
 [target.armv7-unknown-linux-gnueabihf]
 linker = "arm-linux-gnueabihf-gcc"
+
+[target.thumbv7m-none-eabi]
+#runner = 'arm-none-eabi-gdb'
+runner = "gdb-multiarch -q -x openocd.gdb"
```

Comment:
i don't personally use openocd but, it has to be here right (like, there's no way of having this be per-user really)?

you might like to check out [probe-run](https://ferrous-systems.com/blog/probe-run/) (and [defmt](https://ferrous-systems.com/blog/defmt/)) if you haven't come across them yet, a bit of work to get going but a pretty neat dev experience.

Reply:
I think there are alternatives to `openocd`. I started playing a bit with `cargo-embed` https://probe.rs/guide/1_tools/cargo-embed/  and `probe-rs` https://probe.rs/. I'm not sure if/how it is related to `probe-run`. It does not seem to need the `memory.x` file so would remove the need for `build.rs`, but I don't yet understand how it links without a memory map. It seems like there may be a few developments on this front. Would you prefer to just build and not to provide a way to run? Maybe just give some suggestions of ways to do it?

`defmt` I only know through the problems caused by its use of a link, which reduces version flexibility. That got passed through in `bxcan` and breaks multi-HAL testing, see
https://github.com/stm32-rs/stm32f7xx-hal/issues/114#issuecomment-812658708. This has resulted in me having to file several issues and a couple of PRs. Not a very positive introduction, but I should figure out what it does.

Reply:
After looking at this a bit more I think the best approach is to just build and not attempt to provide linking/flashing/debugging capabilities. That avoids committing to changing tools and possibly conflicting with users needs and preferences. The `build.rs` is only to provide the  `memory.x` file for `openocd`. So this would remove the need for the `build.rs` file and the `examples/memoryMaps` directory. The `no_std` sections of `.cargo/config` can either be removed or commented out. (Preferences?)  Comments in `examples/lora_spi_send` would need to be adjusted and might point to different alternatives for linking/flashing/debugging.

I'll go ahead on this if you think it sounds good.